python/taichi/lang/glm.py

```diff
@@ -0,0 +1,92 @@
+import taichi as ti
+
+@ti.func
+def fract(x):
+    x - ti.floor(x)
+
+@ti.func
+def clamp(x, a, b):
+    return ti.min(ti.max(x, a), b)
+
+@ti.func
+def mix(x, y, a):
+    return x * (1 - a) + y * a
+
+@ti.func
+def step(e, x):
+    ret = x
+    if x < e:
+        ret = 0
+    else:
+        ret = 1
+    return ret
+
+@ti.func
+def smoothstep(a, b, x):
+    t = clamp((x - a) / (b - a), 0, 1)
+    return t * t * (3 - 2 * t)
+
+@ti.func
+def sign(x):
+    ret = x
+    if x > 0:
+        ret = 1
+    elif x == 0:
+        ret = 0
+    else:
+        ret = -1
+    return ret
+
+@ti.func
+def exp2(x):
+    return ti.exp(x * ti.ln2)
+
+@ti.func
+def log2(x):
+    return ti.log(x) * ti.log2e
+
+@ti.func
+def vec2(x, y):
+    return ti.Vector([x, y])
+
+@ti.func
+def vec3(x, y, z):
+    return ti.Vector([x, y, z])
+
+@ti.func
+def vec4(x, y, z, w):
+    return ti.Vector([x, y, z, w])
```

Comment:
Not sure how helpful it is to have these... For newcomers these are more functions to learn.

Reply:
I hope I could just `ti.vec(*xs)`, but `@ti.func` doesn't happy with *xs...